src/backend/InvenTree/stock/models.py

```diff
@@ -2085,10 +2091,17 @@ def merge_stock_items(self, other_items, raise_error=False, **kwargs):
         self.save()
 
         # Rebuild stock trees as required
+        rebuild_result = True
         for tree_id in tree_ids:
-            InvenTree.tasks.offload_task(
-                stock.tasks.rebuild_stock_item_tree, tree_id=tree_id, group='stock'
+            if not stock.tasks.rebuild_stock_item_tree(tree_id, rebuild_on_fail=False):
+                rebuild_result = False
```

Comment:
Have you tested what this means for endpoint performance?

Reply:
As the tree issues are now fixed, the rebuild is limited to only the single tree which is "pre-filled" as much as possible. It's quite performant, and does not seem to add significant overhead